src/foam/lib/json/JSONParser.js

```
@@ -51,6 +51,29 @@ foam.CLASS({
         return null;
       }
     }
+
+    public Object[] parseStringForArrayWithException(String data, Class defaultClass) throws Exception {
+      StringPStream ps = new StringPStream();
+      ps.setString(data);
+      ParserContext x = new ParserContextImpl();
+      x.set("X", getX());
+
+      try {
+        ps = (StringPStream) ps.apply(FObjectArrayParser.create(defaultClass), x);
+        if ( ps == null ) {
+          throw new RuntimeException("Parser returned null - failed to parse JSON array");
```

@kgrgreer this is different behaviour than any other parser/parsing (regardless of method naming), and I don't think we want to start down the path of duplicate methods, one which throws, one which does not.   
Perhaps if we provide a different JSONParser (can't think of a good name), that has this behaviour throughout.

---

I do agree that I don't like the duplication of the methods, but don't want to break the other codes :(

maybe we should migrate to functions that doesnt hide the error everywhere?

---

Errors are intentionally ignored so parsing can fall through.  Not ignoring would break a lot of application logic. 
If, for example, a new parser which extends JSONParser, FailingJSONParser, and adds the duplicate methods which throw.   
Or if properties throwExceptions and throwOnNull are added to JSONParser and each method is modified appropriately, then the caller can decide.

---

Added the two props, but lets see kevin input on it

---

looking at foam3/src/foam/core/http/ServiceWebAgent.java, looks like the pattern can be followed by me in upload agent, so going to test it more

---

re: ServiceWebAgent - yes, good idea, forgot about that.